addnode options / loadbalancer stuff

yamins81 opened the pull request:
I added:

```
-- options to addnode and create_nodes
-- ability to balance specified queue to loadbalancer (and various related options)
-- generalized remove_from_sge function in ClusterSetup
-- added sge_utils.py for various SGE utility functions used in several places
```

Commit:
compound node instance types

yamins81 committed:
Merge branch 'master' of git://github.com/jtriley/StarCluster

Conflicts:
	starcluster/cluster.py
	starcluster/config.py
	starcluster/static.py

yamins81 committed:
1) additional node-type arguments to addnode, 2) generalized remove_from_sge, 3) load balancer for specified queue and works for queues with 0 nodes

yamins81 committed:
pep 8 compliance

yamins81 committed:
load balancer takes queue option; add_nodes takes various node options; remove_from_sge generalized; added sge_utils.py module

jtriley commented on starcluster/sge_utils.py:
```diff
@@ -0,0 +1,443 @@
+hostlist NONE
+"""
+
+import BeautifulSoup
```
Do we really need BeautifulSoup here? If possible it'd be nice to use xml.minidom instead since it's included in the standard library. If we _really_ need it we'll need to add BeautifulSoup to the dependency list in setup.py....

yamins81 replied:
No, I'm totally sure that this can be replaced with xml.minidom code ... it was just easier for me since I'm more used to BueaituflSoup.   I'll have alok at replacing it in the next couple of days ...

jtriley replied:
OK, great, thanks!

yamins81 replied:
OK that should be done ... 

I merged stuff from your master as well, there were a couple of conflicts in the loadbalancer ... I think I merged them correctly but  can you go over those to make sure?

yamins81 committed:
replaced beautifulsoup with xml.dom.minidom in sge_utils.get_qstat
Also: Merge branch 'master' of git://github.com/jtriley/StarCluster

Conflicts:
	starcluster/balancers/sge/__init__.py

yamins81 committed:
multple nodes option added to addnode command

yamins81 committed:
Merge branch 'upstream/master'

yamins81 committed:
bugfix addnode

canesin commented:
Hi, I have seen that the pull don't apply cleanly .. how is the progress of adding it to the master or develop branch ??

jtriley commented:
@canesin My apologies for the delay, yes this pull request needs to be updated/cleaned in order to be merged into the develop branch. I'm in the process of trying to push out a 0.92.2 release in the next few days so this will have to wait until after the next release.

yamins81 committed:
merge

yamins81 committed:
small bugfixes

yamins81 committed:
either a bugfix or something required for old versions of boto?

yamins81 committed:
boto thing fixed back to origin state

yamins81 committed:
merge

yamins81 committed:
gitignore

yamins81 committed:
bugfix

yamins81 committed:
bugfix merge

yamins81 committed:
merge
